fenicsadapter/fenicsadapter.py

```diff
@@ -109,21 +109,24 @@ def update_coupling_expression(self, coupling_expression, data):
         ----------
         coupling_expression : Object of class dolfin.functions.expression.Expression
             Reference to object of class GeneralInterpolationExpression or ExactInterpolationExpression.
-        data : array_like
-            The coupling data. A numpy array [N x D] where N = number of vertices and D = dimensions of the data, i.e.
-            for scalar valued data D = 1 and for vector valued data D = dimension of problem.
+        data : dict_like
+            The coupling data. A dictionary containing nodal data with vertex coordinates as key and associated data as
+            value.
         """
-        coupling_expression.update_boundary_data(data, self._coupling_mesh_vertices[:, 0], self._coupling_mesh_vertices[:, 1])
+        x_coord = np.array([vertex[0] for vertex in data.keys()])
+        y_coord = np.array([vertex[1] for vertex in data.keys()])
+        nodal_data = np.array([data for data in data.values()])
+        coupling_expression.update_boundary_data(nodal_data, x_coord, y_coord)
```

Comment:
I think we should also add some checks here (dimension fitting? coupling mesh vertices are still present in data?...)